### ISSUE TITLE
usergroup management with meshctrl 

### ISSUE THREAD

**Issue by guerby** [Ylianst/MeshCentral#1869]
Hi,

To manage user groups with meshctrl more fully, currently there is:

AddUserGroup
RemoveUserGroup

To be able to script efficiently it would be nice to have as well ways to list, add and remove users from UserGroup, ie:

AddUserToUserGroup
RemoveUserFromUserGroup 
RemoveAllUsersFromUserGroup
ListUsersOfUserGroup

Thanks!

**Comment by Ylianst** [comment]
Good request, makes sense.

**Comment by Ylianst** [comment]
Just added all the commands you wanted to MeshCtrl.js. It will be in MeshCentral v0.6.57 when it's next published.

Ylian

```
MeshCtrl performs command line actions on a MeshCentral server.
Information at: https://meshcommander.com/meshcentral
No action specified, use MeshCtrl like this:

  meshctrl [action] [arguments]

Supported actions:
  Help [action]               - Get help on an action.
  ServerInfo                  - Show server information.
  UserInfo                    - Show user information.
  ListUsers                   - List user accounts.
  ListUserSessions            - List online users.
  ListUserGroups              - List user groups.
  ListDevices                 - List devices.
  ListDeviceGroups            - List device groups.
  ListUsersOfDeviceGroup      - List the users in a device group.
  ListUsersOfUserGroup        - List the users in a user group.    <----------
  DeviceInfo                  - Show information about a device.
  Config                      - Perform operation on config.json file.
  AddUser                     - Create a new user account.
  RemoveUser                  - Delete a user account.
  AddUserGroup                - Create a new user group.
  RemoveUserGroup             - Delete a user group.
  AddUserToUserGroup          - Add a user to an existing user group.    <----------
  RemoveUserFromUserGroup     - Remove a user from a user group.    <----------
  RemoveAllUsersFromUserGroup - Remove all users from a user group.    <----------
  AddDeviceGroup              - Create a new device group.
  RemoveDeviceGroup           - Delete a device group.
  EditDeviceGroup             - Change a device group values.
  MoveToDeviceGroup           - Move a device to a different device group.
  AddUserToDeviceGroup        - Add a user to a device group.
  RemoveUserFromDeviceGroup   - Remove a user from a device group.
  AddUserToDevice             - Add a user to a device.
  RemoveUserFromDevice        - Remove a user from a device.
  SendInviteEmail             - Send an agent install invitation email.
  GenerateInviteLink          - Create an invitation link.
  Broadcast                   - Display a message to all online users.
  ShowEvents                  - Display real-time server events in JSON format.
  RunCommand                  - Run a shell command on a remote device.
  Shell                       - Access command shell of a remote device.
  Upload                      - Upload a file to a remote device.
  Download                    - Download a file from a remote device.
  DeviceOpenUrl               - Open a URL on a remote device.
  DeviceMessage               - Open a message box on a remote device.
  DeviceToast                 - Display a toast notification on a remote device.

Supported login arguments:
  --url [wss://server]        - Server url, wss://localhost:443 is default.
                              - Use wss://localhost:443?key=xxx if login key is required.
  --loginuser [username]      - Login username, admin is default.
  --loginpass [password]      - Login password.
  --token [number]            - 2nd factor authentication token.
  --loginkey [hex]            - Server login key in hex.
  --loginkeyfile [file]       - File containing server login key in hex.
  --logindomain [domainid]    - Domain id, default is empty, only used with loginkey.
  --proxy [http://proxy:1]    - Specify an HTTP proxy.
```

**Comment by guerby** [comment]
Thanks!

I think I forgot Remove/AddUserGroupToDevice to be complete (sorry about that !)

**Comment by Ylianst** [comment]
Arg. Ok, just published MeshCentral v0.6.57 with improved MeshCtrl.js The "ListUserGroups" command now shows the user groups and all included users, devices and device groups. Also, to avoid having too many commands, "AddToUserGroup" and "RemoveFromUserGroup" can be used to add/remove users, devices and device groups. Let me know if that works.

```
MeshCtrl performs command line actions on a MeshCentral server.
Information at: https://meshcommander.com/meshcentral
No action specified, use MeshCtrl like this:

  meshctrl [action] [arguments]

Supported actions:
  Help [action]               - Get help on an action.
  ServerInfo                  - Show server information.
  UserInfo                    - Show user information.
  ListUsers                   - List user accounts.
  ListUserSessions            - List online users.
  ListUserGroups              - List user groups.    <----------
  ListDevices                 - List devices.
  ListDeviceGroups            - List device groups.
  ListUsersOfDeviceGroup      - List the users in a device group.
  DeviceInfo                  - Show information about a device.
  Config                      - Perform operation on config.json file.
  AddUser                     - Create a new user account.
  RemoveUser                  - Delete a user account.
  AddUserGroup                - Create a new user group.
  RemoveUserGroup             - Delete a user group.
  AddToUserGroup              - Add a user, device or device group to a user group.    <----------
  RemoveFromUserGroup         - Remove a user, device or device group from a user group.    <----------
  RemoveAllUsersFromUserGroup - Remove all users from a user group.
  AddDeviceGroup              - Create a new device group.
  RemoveDeviceGroup           - Delete a device group.
  EditDeviceGroup             - Change a device group values.
  MoveToDeviceGroup           - Move a device to a different device group.
  AddUserToDeviceGroup        - Add a user to a device group.
  RemoveUserFromDeviceGroup   - Remove a user from a device group.
  AddUserToDevice             - Add a user to a device.
  RemoveUserFromDevice        - Remove a user from a device.
  SendInviteEmail             - Send an agent install invitation email.
  GenerateInviteLink          - Create an invitation link.
  Broadcast                   - Display a message to all online users.
  ShowEvents                  - Display real-time server events in JSON format.
  RunCommand                  - Run a shell command on a remote device.
  Shell                       - Access command shell of a remote device.
  Upload                      - Upload a file to a remote device.
  Download                    - Download a file from a remote device.
  DeviceOpenUrl               - Open a URL on a remote device.
  DeviceMessage               - Open a message box on a remote device.
  DeviceToast                 - Display a toast notification on a remote device.

Supported login arguments:
  --url [wss://server]        - Server url, wss://localhost:443 is default.
                              - Use wss://localhost:443?key=xxx if login key is required.
  --loginuser [username]      - Login username, admin is default.
  --loginpass [password]      - Login password.
  --token [number]            - 2nd factor authentication token.
  --loginkey [hex]            - Server login key in hex.
  --loginkeyfile [file]       - File containing server login key in hex.
  --logindomain [domainid]    - Domain id, default is empty, only used with loginkey.
  --proxy [http://proxy:1]    - Specify an HTTP proxy.
```
